spark-etl/src/main/scala/geotrellis/spark/etl/Etl.scala

```diff
@@ -0,0 +1,64 @@
+package geotrellis.spark.etl
+
+import geotrellis.spark.ingest.Pyramid
+import geotrellis.spark.io.index.KeyIndexMethod
+import geotrellis.spark.{SpatialComponent, RasterRDD, LayerId}
+import geotrellis.spark.tiling.{LayoutLevel, ZoomedLayoutScheme}
+import com.google.inject._
+import org.apache.spark.SparkContext
+import scala.reflect._
+import scala.collection.JavaConverters._
+
+object Etl {
+  def apply[K: ClassTag: SpatialComponent](args: Seq[String]): Etl[K] = {
+    new Etl(args, s3.S3Module, hadoop.HadoopModule)
+  }
+}
+
+case class Etl[K: ClassTag: SpatialComponent](args: Seq[String], modules: Module*) {
+  val conf = new EtlConf(args)
+
+  val (ingestPlugin, outputPlugin) = {
+    val injector = Guice.createInjector(modules: _*)
+
+    val ingest = injector
+      .findBindingsByType(new TypeLiteral[IngestPlugin]{})
+      .asScala
+      .map { _.getProvider.get }
+      .find { _.suitableFor(conf.ingest(), conf.format(), classTag[K]) }
+      .getOrElse(sys.error(s"Unable to find ingest module of type '${conf.ingest()}' for format `${conf.format()} and key ${classTag[K]}"))
+
+    ingest.validate(conf.ingestProps)
+
+    val output = injector
+      .findBindingsByType(new TypeLiteral[SinkPlugin]{})
+      .asScala
+      .map { _.getProvider.get }
+      .find { _.suitableFor(conf.output(), classTag[K]) }
+      .getOrElse(sys.error(s"Unable to find output module of type '${conf.output()}' and key ${classTag[K]}"))
+
+    output.validate(conf.outputProps)
+
+    (ingest, output)
+  }
+
+  def ingest()(implicit sc: SparkContext): (LayerId, RasterRDD[K]) = {
+    val (level, rdd) = ingestPlugin[K](conf.cache(), conf.crs(), ZoomedLayoutScheme(conf.tileSize()), conf.ingestProps)
+    LayerId(conf.layerName(), level.zoom) -> rdd
+  }
+
+  def save(id: LayerId, rdd: RasterRDD[K], method: KeyIndexMethod[K]): Unit = {
```

Comment:
@lossyrob does this become "output" ?
I'd like to suggest naming load/save to keep functions as verbs vs nouns.